Feature turnbased battle

MiniBunnyFooFoo opened the pull request:
SETUP FOR NEW RUNNING ENVIRONMENT:
cd to `cd code/beastly-brawl-showdown`
Follow all steps [from this website](https://www.mongodb.com/docs/manual/tutorial/install-mongodb-on-ubuntu/) in the "Install MongoDB Community Edition" tab from step 1- 4
[Download typescript and ts-node](https://typestrong.org/ts-node/docs/installation/) if you haven't already

**IF YOU ARE GETTING THIS ERROR:**
![image](https://github.com/user-attachments/assets/0bcd14ff-9e68-4ab8-9dbb-a51122644329)
`cd code/beastly-brawl-showdown`
`meteor npm install`

**IF YOU ARE STILL GETTING ERRORS**
`cd code/beastly-brawl` (if not already)
`npm install *INSERT PACKAGE HERE*` (express, socket.io, whatever you are missing)

In one terminal, run: 
`cd code`
`sh start_db.sh`
and to start the server
`cd code/game-server-v2`
`ts-node main.ts`

in another terminal
cd to `code/beastly-brawl-showdown`
`sh run.sh` to run the application

MiniBunnyFooFoo committed:
fixed major bug, socketID should not be used as persistent identifier

MiniBunnyFooFoo committed:
brought in old changes from previous turnbased branch

MiniBunnyFooFoo committed:
shows monster selection over the player page before navigation to battle

Geartown08 committed:
Rearranged monster images to work with the new Monster classes

MiniBunnyFooFoo committed:
Setup monsters to be assigned to players

Geartown08 committed:
Made it so that the confirm button in the monsterselection screen goes to the Battle screen

Geartown08 committed:
Removed some unused variables and assigned some value types

Geartown08 committed:
removed more unused variables to help pass the pipeline

Goonta12 reviewed (commented):
No internal doc in some areas. However because there are comments in the most crucial / complex areas, acceptable for now.

jamiekaltzis commented on code/beastly-brawl-showdown/imports/data/monsters/Monsters.ts:
Monsters are currently identified by type which is not very distinguishable. A name attribute will need to be added to distinguish between different monsters of the same type.

Mechanised-Mango-Box committed:
Moved monster selection into player page

Mechanised-Mango-Box committed:
Added temp - interact  (but doesnt do anything)

Skygeneral133 committed:
fixed problem where battlescreens formating was off

Skygeneral133 committed:
fixed a bug with the displays

Skygeneral133 committed:
fixed all ui bugs

Mechanised-Mango-Box committed:
Added todo comments

- note that these are genral guidelines
- emit and on may not match up right now - ensure you fix this

Henryalditio committed:
Added temporary match class and additions to room class.

Henryalditio committed:
Created method to create matches and get them based off of matchid

Geartown08 committed:
Added broadcasts and props where it seems appropriate, focused on Battle Screen, Actions and Monsters

Geartown08 committed:
Fixed some errors that changes made to MonsterData affected

Henryalditio committed:
Uploading current temporary implementation

Henryalditio committed:
Merge branch 'feature-turnbased_battle' of https://github.com/Monash-FIT3170/2025W1-Beastly-Brawl-Showdown into feature-turnbased_battle

Henryalditio committed:
Right now, matches are assigned to every player in a room when "create-matches" event occurs.

weihom05 committed:
Added opaque to show a more clearer indication of a monster being selected

Skygeneral133 committed:
added initial choose mattchup logic

Skygeneral133 committed:
made it so enemy is returned to the playerpage after everybody has selected mosnters

Skygeneral133 committed:
added comments and fixed a bug

HarshilPatel69 committed:
Updated match to pass data once match starts to players

HarshilPatel69 committed:
Merge branch 'feature-turnbased_battle' of https://github.com/Monash-FIT3170/2025W1-Beastly-Brawl-Showdown into feature-turnbased_battle

Henryalditio committed:
Added some helper functions for battle logic.

MiniBunnyFooFoo committed:
restore run.sh that was deleted in merge

MiniBunnyFooFoo committed:
amended run.sh, game can now start

MiniBunnyFooFoo committed:
fix: monster selection now returns to playerpage with a state

MiniBunnyFooFoo committed:
updated comments and names for readability

Geartown08 committed:
Made the Player notify the server about their choice of monster and then server broadcasts its ready for the battle stage

Skygeneral133 committed:
added back my work

Mechanised-Mango-Box committed:
Tighten Type Safety

Geartown08 committed:
Made it so you can go from Monster Selection to Battle Screen

MiniBunnyFooFoo committed:
Merge branch 'feature-turnbased_battle' into fix-seperate_player_screen_components

Mechanised-Mango-Box committed:
Projector screen no lnger refreshes connection on react render pass

MiniBunnyFooFoo committed:
added comments to PlayerPage.tsx to show how it works

MiniBunnyFooFoo committed:
updated package lock and added TODOs for action classes

Mechanised-Mango-Box committed:
Remove duplicate event

Mechanised-Mango-Box committed:
Player given room id (since players exist only per room)

Mechanised-Mango-Box committed:
Format files

Mechanised-Mango-Box committed:
Added comments (see word map)

Mechanised-Mango-Box committed:
Simplify Match data

Mechanised-Mango-Box committed:
Fixed function name

Mechanised-Mango-Box committed:
Fixed function name (again)

MiniBunnyFooFoo committed:
Added mark's match start and monster selection emits (few more to go)

Mechanised-Mango-Box committed:
Fixed Type errors

MiniBunnyFooFoo committed:
extra comments for later setup

Mechanised-Mango-Box committed:
Merge branch 'feature-turnbased_battle' into fix-seperate_player_screen_components

Mechanised-Mango-Box committed:
Merge pull request #31 from Monash-FIT3170/fix-seperate_player_screen_components

fix: monster selection now returns to playerpage with a state

Goonta12 committed:
Added a listener for the buttons on battle screen.

Mechanised-Mango-Box committed:
Add match generator which accounts for odd participants (byes system) - also fixed up match types

Geartown08 committed:
Made it so that Battle Screen saves the PlayerSocket when the player switches between screens

Goonta12 committed:
Implemented the abilty for players to send chosen action to server.

Goonta12 committed:
Merge branch 'feature-turnbased_battle' into feature-listener_for_battle

Goonta12 committed:
Changed Main.

Mechanised-Mango-Box committed:
Fixed error does not stop match type check causing program to fall through to error case

Geartown08 committed:
Fixed the BattleScreen being unable to load images

Goonta12 committed:
Merge branch 'feature-turnbased_battle' into feature-listener_for_battle

MiniBunnyFooFoo committed:
server now awaits all players selecting monsters before transitioning to battle screen

MiniBunnyFooFoo committed:
Merge branch 'feature-turnbased_battle' of https://github.com/Monash-FIT3170/2025W1-Beastly-Brawl-Showdown into feature-turnbased_battle

Mechanised-Mango-Box committed:
Merge branch 'feature-turnbased_battle' into feature-listener_for_battle

Mechanised-Mango-Box committed:
Merge pull request #33 from Monash-FIT3170/feature-listener_for_battle

Feature listener for battle

Geartown08 committed:
Removing unecessary files

Geartown08 committed:
Added the functionality for a match to start between two players, data of the match is kept on a log file that can be updated

Geartown08 committed:
Created an initial design for the buttons to register the user's inputs for attack, defend and special

Geartown08 committed:
Made it so that it skips the part of writing the file and the monster name is sent directly to the player in order to display them both

MiniBunnyFooFoo committed:
ryans computer was crying over this file so i deleted it

Geartown08 committed:
Added the emit and on for sockets so that a player's move is registered and read by the server before the calculations are done for the move

Geartown08 committed:
Changed the Monster Selection in Server Main so that it handles more than just 2 players and then each player is set to their own seperate Match against another player

Geartown08 committed:
Made it so that after a two players are sent to a room, then then an action can be selected and that is reflected on the server and then the battle is updated and is sent back to the player and then waiting for a new action. Fixed a bug with match creation and also have a bug with updating the monster's health so waiting on the new monster files first

Mechanised-Mango-Box commented:
If this is no longer needed close this. @Geartown08 @MiniBunnyFooFoo

Mechanised-Mango-Box commented:
Refer to Simulator
